### ISSUE TITLE
Google Analytics database connection stopped working.

### ISSUE THREAD

**Issue by rodolphogarrido** [metabase/metabase#14578]
**Describe the bug**
Google Analytics questions stopped working displaying an error of "bad_request". I've got suspicious it could be an issue with the connection and tried to create a new access token, but when I try to use it I get an error saying "Request had invalid authentication credentials" (see screenshot) but oddly I'm able to create a new Google Analytics database connection using the exact credential and access token without problems and use this new database. My issue here is that all questions based on the problematic database connection can't be used.

**Expected behavior**
Be able to change Google analytics database connection credentials and use it.

**Screenshots**
![ga_problem](https://user-images.githubusercontent.com/47122489/106214433-87c4f300-61ad-11eb-97a2-5d2aae5846e8.png)

**Severity**
Users can't see queries created from Google Analytics nor use it (300+ questions).

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "5.4.72-flatcar",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres",
      "presto",
      "googleanalytics"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.5"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-01-20",
      "tag": "v0.37.7",
      "branch": "release-x.37.x",
      "hash": "2b034aa"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

**Comment by flamber** [comment]
Hi @rodolphogarrido 
Very strange. I have no idea what could have caused this, but in the application database (Postgres for you), have a look in the table `metabase_database`. I would compare the working and broken GA - specifically `details` column. You should be able to copy the working to the broken, and then remove the working, since the broken should now be "working" (too many reference? :laughing:)
You should of course never touch application database until you have made a backup, so you can revert in case you break everything even more.
While I would love to figure out the root cause of this, since it would allow us to create a fix, I think it's going to be very difficult to figure out what has happened.

**Comment by rodolphogarrido** [comment]
Hi @flamber 
I appreciate your advice very much!  The explanation was perfectly clear 😄! I was able to recover the "broken" connection by copying the value of the field "details" from the "working" connection as you advised.

I understand it might be almost impossible to figure out what caused this.

Thanks for the help 😄

**Comment by flamber** [comment]
@rodolphogarrido Could you spot the problem between the working and broken `details`? I know you cannot post tokens, but I'm interested if the structure was different or if anything but the `access/refresh-token` was different.

**Comment by rodolphogarrido** [comment]
@flamber Sorry I took this long to answer. At the time I didn't spot the problem between the working and broken `details`. Only the `access/refresh-token` was different.

**Comment by janmpeterka** [comment]
Not sure if this is related, but it might be:

Our Google Analytics connection stops working periodically.
We were unable to solve it any other way than deleting db and creating new one, until today.
Now we found a sort-of solution:
There was problem with refresh-token not working, and also (probably) bug when we couldn't insert new auth token.
When we deleted refresh-token from `metabase.metabase_databases`, we were allowed to insert auth token again in db settings, and new, working refresh-token was generated.
